packages/ui/src/views/repo/repo-layout/index.tsx

```diff
@@ -8,10 +8,18 @@ export const RepoLayout = ({ useTranslationStore }: { useTranslationStore: () =>
   const activeTab = location.pathname.split('/').pop() || 'summary'
   const { t } = useTranslationStore()
 
+  // TODO: should have a more robust function that will determine tab since there can be nested states
+  const getFinalTab = () => {
```

Comment:
can we use `useMemo` here instead of using function?

Reply:
done @praneshg239